### PR TITLE
feat(TDOPS-14/FolderTreeView): added disabled list item state

### PR DESCRIPTION
**Pull request by vyshkov**
**What is the problem this PR is trying to solve?**

The common component should be updated:

- to update the style of the active state, hover state
- and introduce the disabled state style

More details: https://jira.talendforge.org/browse/TDOPS-14
Link to frontify: https://company-57688.frontify.com/document/92132#/ui-controls/tree-view

**What is the chosen solution to this problem?**

Fixed css

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/components/src/TreeView/TreeViewItem/TreeViewItem.component.js [hunk 1]
```diff
@@ -213,6 +214,7 @@ class TreeViewItem extends React.Component {
 			actions,
 			icon,
 			counter = children.length,
+			disabled,
```
Comment (frassinier) [inline packages/components/src/TreeView/TreeViewItem/TreeViewItem.component.js:217]: `disabled` must be passed to the Action, isn't it?

Comment (vyshkov) [reply packages/components/src/TreeView/TreeViewItem/TreeViewItem.component.js:217]: I thought about it, disabling all actions limits our flexibility.  We can set actions to be disabled using action props, and with this approach, disabled list items may contain other clickable actions (info, warning etc) that may be better for the user experience.

Comment (jmfrancois) [reply packages/components/src/TreeView/TreeViewItem/TreeViewItem.component.js:217]: The onselect action is still fired on the demo so I don t understand it s only visual disabled right ?

**Review by jmfrancois — CHANGES_REQUESTED** [review]
The definition said:
When an element cannot be selected the whole line is 0.54 opacity, with a not-allowed cursor.
But the props disabled only apply css style it do not prevent the onSelect to be triggered by a click event

File: packages/components/src/TreeView/TreeViewItem/TreeViewItem.component.js (hunk 1 above)
Comment (vyshkov) [reply packages/components/src/TreeView/TreeViewItem/TreeViewItem.component.js:217]: Yes, you are right. We can also skip onSelect callback for disabled items. But onToggle should be available for the user. We use it on plan rerun dialog, and the user should be able to see what's inside disabled steps.

**Review by jmfrancois — APPROVED** [review]

File: packages/components/src/TreeView/TreeViewItem/TreeViewItem.component.js (hunk 1 above)
Comment (frassinier) [reply packages/components/src/TreeView/TreeViewItem/TreeViewItem.component.js:217]: Using the keyboard I can still fire the onSelect on the demo, so we must consider modifying the other `onKey*`handlers.

Another thing, we need to get the info that this button is actually `[aria-disabled=true]` if we want to manage it manually!

Comment (vyshkov) [reply packages/components/src/TreeView/TreeViewItem/TreeViewItem.component.js:217]: good catch @frassinier !

**Review by frassinier — APPROVED** [review]

**Pull request merged**